### PR TITLE
Add Kubelet server metrics for response status #95307

### LINKED ISSUE #95307: Add Kubelet server metrics for response status
The kubelet exposes some metrics HTTP server metrics: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/metrics/metrics.go

However, none of the metrics include response status. Response status metrics are useful for identifying authorization problems, or malformed client requests.

/sig instrumentation
/sig node
/help

### PR DESCRIPTION
**Pull request by itonyli**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
See #95307

**Which issue(s) this PR fixes**:
Fixes #95307

Fixes #

**Special notes for your reviewer**:
@tallclair @ehashman

**Does this PR introduce a user-facing change?**:

```release-note
Add Kubelet server metrics for response status 
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: pkg/kubelet/server/server.go [hunk 1]
```diff
@@ -938,18 +957,18 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
...
 	method, path := s.getMetricMethodBucket(req.Method), s.getMetricBucket(req.URL.Path)
-
 	longRunning := strconv.FormatBool(isLongRunningRequest(path))
 
-	servermetrics.HTTPRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
+	startTime := time.Now()
 
-	servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
-	defer servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Dec()
+	handler.ServeHTTP(&rw, req)
```
Comment (cheftako) [inline pkg/kubelet/server/server.go:964]: Moving ServeHTTP in relation to the metric calls means we are fundamentally changing the behavior in the event of an exception. I think it is better to record the metrics about the incoming problematic request in that case.

**Comment by ITonyLi** [comment]
I dont think HTTPInflightRequests.WithLabelValues will throw exception. Now，What happend when throw exception on the defer block execute？


------------------ Original ------------------

File: pkg/kubelet/server/server.go (hunk 1 above)
Comment (ITonyLi) [reply pkg/kubelet/server/server.go:964]: When record the metrics about the incoming problematic request, Can`t record the http response status code. So i am Moving ServeHTTP in relation to the metric calls.

Comment (ITonyLi) [reply pkg/kubelet/server/server.go:964]: @tallclair, what do you think of this code change

Comment (cheftako) [reply pkg/kubelet/server/server.go:964]: We could probably guess the status code if ServeHTTP throws an exception (500) but I agree we cannot be sure.  I'm not sure that not knowing the status code of the response is a good enough reason to not record that we got the call.

**Comment by ehashman** [comment]
This PR appears to be trying to solve the same thing as #95429 but slightly differently, is there a reason for the differing implementation?

**Review by itonyli — COMMENTED** [review]
> This PR appears to be trying to solve the same thing as #95429 but slightly differently, is there a reason for the differing implementation?

I think this is a better way to implement and covers three statistical methods.

**Comment by ehashman** [comment]
> I think this is a better way to implement and covers three statistical methods.

Can you please provide a little more detail as to why you think this is a better way and comment on the PR I linked above? Thanks!

**Comment by ITonyLi** [comment]
> > I think this is a better way to implement and covers three statistical methods.
> 
> Can you please provide a little more detail as to why you think this is a better way and comment on the PR I linked above? Thanks!

First, the implementation includes three types of statistics about HTTP status code.
Second, change small, the implementation will not interfere with the operation of other code, such as intrusion and change of httplog code block.

This is my idea. What do you think? Thank you for your attention @ehashman

**Pull request closed without merging**